### PR TITLE
Clean up builds

### PR DESCRIPTION
**Pull request by edgchen1**
**Description**
Update training Python packaging build to use get_docker_image.py.
Remove BUILD_EXTR_PAR docker build argument.
Update get_docker_image.py to check again for the image in the cache after building and before pushing to reduce the chance of a redundant push.

**Motivation and Context**
Use build image cache, reduce number of cached images (previously different values of BUILD_EXTR_PAR would result in distinct cached images).

### CODE REVIEW

**Comment by baijumeswani** [comment]
Might need to also update [tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-e2e-test-nightly-pipeline.yml](https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-e2e-test-nightly-pipeline.yml#L39) so that it passes in the ```-m``` argument for testing since that pipeline depends on openmpi being installed.

**Review by liqunfu — APPROVED** [review]

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_gpu [hunk 1]
```diff
@@ -25,14 +25,14 @@ ENV SSL_CERT_FILE=/opt/_internal/certs.pem
 
 #Build manylinux2014 docker image end
 
-ARG PYTHON_VERSION=3.5
-ARG BUILD_EXTR_PAR
+ARG PYTHON_VERSION=3.6
+ARG INSTALL_DEPS_EXTRA_ARGS
```
Comment (snnn) [inline tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_gpu:29]: As we plan to remove python 3.5 support soon, this is fine.

**Review by snnn — APPROVED** [review]

**Review by baijumeswani — COMMENTED** [review]
Looks good.

**Review by baijumeswani — APPROVED** [review]

File: tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_gpu (hunk 1 above)
Comment (snnn) [reply tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_gpu:29]: Wait, why do we need this argument in the first place? Should the image support all the python versions?

Comment (snnn) [reply tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_gpu:29]: Oh, it is only for training.

Comment (edgchen1) [reply tools/ci_build/github/linux/docker/Dockerfile.manylinux2014_gpu:29]: it's a manylinux image with multiple python versions, but this parameter is used to set the python version for which the dependencies are installed.
yes, this one is only for training. ideally at some point we can use the same image

**Pull request merged**